### PR TITLE
[Design System] Radio Button Component

### PR DESCRIPTION
**Pull request by levadadenys**
### [Design System ] Radio Button Component
* implement states of radio button
* implement sizes of radio button
* added RadioButtonsGroup

Documentation to be done/added separately from this PR.

Video of Storybook:

https://github.com/code-dot-org/code-dot-org/assets/22244040/b8fa98ca-3c42-4dbc-b02e-add1db9454bb

Scrrenshot of Figma:
image

## Links

- jira ticket: [DES-291](https://codedotorg.atlassian.net/browse/DES-291?atlOrigin=eyJpIjoiYjY2NDU3YzllMThkNDg4MDhiYTk2MmNjN2I0NDdiMjIiLCJwIjoiaiJ9)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/componentLibrary/radioButton/radioButton.module.scss [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  display: flex;
+  gap: 10px;
+  align-items: center;
+
+  input[type="radio"] {
+    position: absolute;
+    opacity: 0;
+  }
+
+  i {
```
Comment (sanchitmalhotra126) [inline apps/src/componentLibrary/radioButton/radioButton.module.scss]: Can these rules be applied via classnames?

Comment (levadadenys) [reply apps/src/componentLibrary/radioButton/radioButton.module.scss]: Agree, it might improve readability of styles. Updated

**Comment by levadadenys** [comment]
* added RadioButtonsGroup
Should make easier for devs to render multiple radioButtons in one group, has basic logic that can be extended via external onChange prop

File: apps/src/componentLibrary/radioButton/RadioButton.story.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import {RadioButton, RadioButtonsGroup} from './index';
+
+export default {
```
Comment (fisher-alice) [inline apps/src/componentLibrary/radioButton/RadioButton.story.jsx:4]: Just wanted to leave a note that I checked out your branch to view the story and it looks great! Nice job!

**Review by fisher-alice — APPROVED** [review]
Nice job on this!

File: apps/src/componentLibrary/radioButton/RadioButton.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          visualAppearance="body-one"
+        >
+          {label}
+        </Typography>
+      )}
+    </label>
+  );
+};
+
+export default memo(RadioButton);
```
Comment (sanchitmalhotra126) [inline apps/src/componentLibrary/radioButton/RadioButton.tsx:56]: Great reminder to use `memo` where possible!

**Review by sanchitmalhotra126 — APPROVED** [review]

**Pull request merged**